webapp/src/app/shared/widgets/number-widget/number-widget.component.ts

```diff
@@ -1,14 +1,38 @@
-import {Component, OnInit} from '@angular/core';
+import {Component, OnInit, ViewChild, ElementRef} from '@angular/core';
 import {AbstractWidget} from '../abstract-widget';
 
 @Component({
 	selector: 'app-number-widget',
 	templateUrl: './number-widget.component.html',
 	styleUrls: ['./number-widget.component.scss', '../widget.scss']
 })
-export class NumberWidgetComponent extends AbstractWidget {
-	constructor() {
-		super();
-	}
+export class NumberWidgetComponent extends AbstractWidget implements OnInit {
+	@ViewChild('numberInput', {static: true}) input!: ElementRef;
+
+	private lastValidValue = '0';
+	
+	setSettings(key: string, event: any) {
+		
+
+		let value = event.target.value;
+		
+		if (event.data) {
+			if (event.data.length > value.length) {
+				// input property sets
+				// the value to "" when it
+				// has an invalid input
+				value = this.lastValidValue;	
+			}
+		}
+
+		// Number converts "" to 0
+		// while parseFloat converts "" to NaN
+		const num = Number(value);
 
+		this.lastValidValue = num.toString();
+
+		this.input.nativeElement.value = num.toString();
```

Comment:
There is no need to get the input element as a ViewChild. ngModel on the html element handles the value and it should not be overriden by directly accessing the native element.

Reply:
There is a need when I need to control the input.

Reply:
angular does not behave how I want it to if the same value is set twice.